src/context.c

```diff
@@ -274,12 +274,11 @@ static GHashTable *get_system_info_from_handler(GError **error)
 		return NULL;
 	}
 
-	g_clear_pointer(&context->system_serial, g_free);
-	g_clear_pointer(&context->config->system_variant, g_free);
 	g_hash_table_iter_init(&iter, vars);
 	while (g_hash_table_iter_next(&iter, (gpointer*) &key, (gpointer*) &value)) {
 		/* legacy handling */
 		if (g_strcmp0(key, "RAUC_SYSTEM_SERIAL") == 0) {
+			g_clear_pointer(&context->system_serial, g_free);
```

Comment:
I'd prefer to use `g_clear_pointer` in both branches to keep them consistent. That could be done in the same commit.

Reply:
Sounds good. I'm on vacation right now and unable to build a test case at the moment, but will make the modification you suggested.